internal/services/containerapps/container_app_job_resource.go

```diff
@@ -93,15 +98,93 @@ func (r ContainerAppJobResource) Arguments() map[string]*schema.Schema {
 
 		"template": helpers.JobTemplateSchema(),
 
-		"secrets": helpers.SecretsSchema(),
+		"secret": helpers.SecretsSchema(),
```

Comment:
This schema probably wants modifying to to conditionally `ConflictWith` `secrets`

Reply:
@stephybun I did think about that - but it would mean duplicating the contents of the function (which is currently shared with `azurerm_container_app`) for the remaining duration of the life of 3.0. I'm happy to do the duplication, with a note to go back to the function once 4.0 is released. Which is less work for the team?

Reply:
The duplication with a note for post 4.0 would be fine for us here